[terra-form-select] Combobox custom placeholder remove (#3953)
cerner/terra-core: packages/terra-form-select/CHANGELOG.md

## Unreleased

> *Fixed
>   *Fixed the `combobox` won't take custom placeholder text and only string `-Select- ` will be shown.

Reviewer: ```suggestion
* Changed
  * Removed support for custom placeholder on the `combobox`
```

Reviewer 2: Updated with title "Breaking change" as per below comment,
